docs/tutorial/rewriter/rewrite_patterns.md

```diff
@@ -152,6 +152,20 @@ In order to apply this method to the example above, first create the two separat
 :pyobject: erf_gelu_pattern_2
 ```
 
+:::{note}
+:name: rule-application-order-matters
+
+### Order of Rewrite Rules Matters
+
+When you pass multiple rules in `pattern_rewrite_rules`, the **order in which they appear is important**. 
+
+This is because some rules may depend on patterns created or modified by earlier rules. For example, if `rule1` can only match after `rule2` has made a specific change in the model, then `rule2` must come **before** `rule1` in the list.
```

Comment:
```suggestion
This is because some rules may depend on patterns created or modified by earlier rules. For example, if `rule2` can only match after `rule1` has made a specific change in the model, then `rule1` must come **before** `rule2` in the list.
```

nit: slightly less cognitive load